packages/task/src/browser/task-schema-updater.ts

```diff
@@ -49,17 +72,14 @@ export class TaskSchemaUpdater {
         };
         const taskTypes = await this.taskService.getRegisteredTaskTypes();
         taskSchema.properties.tasks.items.oneOf![0].allOf![0].properties!.type.enum = taskTypes;
-        const taskSchemaUri = new URI(taskSchemaId);
-        const contents = JSON.stringify(taskSchema);
-        try {
-            this.inmemoryResources.update(taskSchemaUri, contents);
-        } catch (e) {
-            this.inmemoryResources.add(taskSchemaUri, contents);
-            this.jsonSchemaStore.registerSchema({
-                fileMatch: ['tasks.json'],
-                url: taskSchemaUri.toString()
-            });
-        }
+        return JSON.stringify(taskSchema);
+    }
+
+    private updateProblemMatcherNames(): void {
```

Comment:
@elaihau do you mind documenting this method?
I see that we `getAll` named problem matchers, reset the current list, and finally push the new list.
I just wanted to understand why it's required and in what scenario.

Reply:
Yep i will document this function. 

This function is called in the callback of `this.problemMatcherRegistry.onDidChangeProblemMatcher( )`. In other words, every time a problem matcher is registered or disposed, we want to update the task schema to ensure the users get the most up-to-date when they are manually entering the problem matcher(s).

Actually I can simply do `problemMatcherNames = matcherNames;` instead of resetting the length of array.